File: api/pkg/vetchi/employer-auth.go
package vetchi

type EmployerSignInRequest struct {
	ClientID string       `json:"client_id" validate:"required,client_id"`
	Email    EmailAddress `json:"email"     validate:"required,email"`
	Password Password     `json:"password"  validate:"required,password"`
}

type EmployerSignInResponse struct {
	Token string `json:"token"`
}

type EmployerTFARequest struct {
	TFACode    string `json:"tfa_code"              validate:"required"`
	TFAToken   string `json:"tfa_token"             validate:"required"`
	RememberMe bool   `json:"remember_me,omitempty"`
}

type EmployerTFAResponse struct {
	SessionToken string `json:"session_token"`
}

type GetOnboardStatusRequest struct {
	ClientID string `json:"client_id" validate:"required,client_id"`
}

type GetOnboardStatusResponse struct {
	Status OnboardStatus `json:"status"`
}

type OnboardStatus string

const (
	DomainNotVerified            OnboardStatus = "DOMAIN_NOT_VERIFIED"
	DomainVerifiedOnboardPending OnboardStatus = "DOMAIN_VERIFIED_ONBOARD_PENDING"
	DomainOnboarded              OnboardStatus = "DOMAIN_ONBOARDED"
)

type SetOnboardPasswordRequest struct {
	ClientID string   `json:"client_id" validate:"required,client_id"`
	Password Password `json:"password"  validate:"required,password"`
	Token    string   `json:"token"     validate:"required"`
}

File: api/pkg/vetchi/employer-auth.ts
import { EmailAddress, Password } from "./common";

export type OnboardStatus =
  | "DOMAIN_NOT_VERIFIED"
  | "DOMAIN_VERIFIED_ONBOARD_PENDING"
  | "DOMAIN_ONBOARDED";

export interface GetOnboardStatusRequest {
  client_id: string;
}

export interface GetOnboardStatusResponse {
  status: OnboardStatus;
}

export interface SetOnboardPasswordRequest {
  client_id: string;
  password: Password;
  token: string;
}

export interface EmployerSignInRequest {
  client_id: string;
  email: EmailAddress;
  password: Password;
}

export interface EmployerSignInResponse {
  token: string;
}

export interface EmployerTFARequest {
  tfa_code: string;
  tfa_token: string;
  remember_me?: boolean;
}

export interface EmployerTFAResponse {
  session_token: string;
}
